cli/tests/activate.bats

```diff
@@ -4768,3 +4768,17 @@ Setting PATH from .bashrc"
   assert_equal "$stderr" "Sourcing .bashrc
 Setting PATH from .bashrc"
 }
+
+# bats test_tags=activate:upgrade-checks
+@test "runs upgrade checks in the background on activate" {
+    unset _FLOX_TESTING_DISABLE_BG_SIDE_EFFECTS # allow background checks for this test
+    "$FLOX_BIN" init
+    _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/resolve/old_hello.json" "$FLOX_BIN" install hello
+    _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/resolve/hello.json" "$FLOX_BIN" activate -- true
+
+    timeout 5s bash -c "while [ ! -f .flox/cache/upgrade-checks.json ]; do sleep 0.1; done"
```

Comment:
resolution here is instant'ish but we do check if the environment can be built which pulls down hello and in isolation on bad days may trip the timeout.
Since we are ostensibly a hello-package-manager, i dont _think_ this will be an issue (hello should already be there from prior tests).